cmd/snapshot.go

```diff
@@ -48,25 +48,28 @@ storing snapshots on various cloud storage providers as well as local disk locat
 				insecureTransport,
 				insecureSkipVerify,
 				etcdEndpoints)
-			ssr, err := snapshotter.NewSnapshotter(
+			snapshotterConfig, err := snapshotter.NewSnapshotterConfig(
 				schedule,
 				ss,
-				logger,
 				maxBackups,
 				deltaSnapshotIntervalSeconds,
 				time.Duration(etcdConnectionTimeout),
 				time.Duration(garbageCollectionPeriodSeconds),
 				garbageCollectionPolicy,
 				tlsConfig)
 			if err != nil {
-				logger.Fatalf("Failed to create snapshotter: %v", err)
+				logger.Fatalf("failed to create snapstore config: %v", err)
 			}
-			gcStopCh := make(chan bool)
-			go ssr.GarbageCollector(gcStopCh)
-			if err := ssr.Run(false, stopCh); err != nil {
+			ssr := snapshotter.NewSnapshotter(
+				logger,
+				snapshotterConfig)
+
+			gcStopCh := make(chan struct{})
```

Comment:
Isn't section a duplication of code from `cmd/server.go`?

Reply:
Actually yes. but `cmd/snapshot.go` and `cmd/server.go` are two different commands with different parameters to `ssr.Run`. So, we have to duplicate the code. Regarding `ssr.RunGarbageCollelctor`, we wanted it to separate from  `ssr.Run` so explicitly call it, instead of calling it from `ssr.Run`